### PR TITLE
SAK-46566 BLTI content item (not tool) secret keys should be encrypted

### PR DESCRIPTION
**Pull request by kunaljaykam**
https://sakaiproject.atlassian.net/browse/SAK-46566

### CODE REVIEW

**Comment by csev** [comment]
Thanks for the revisions - much easier to review :) Thanks.

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java [hunk 1]
```diff
@@ -1452,6 +1456,15 @@ private void doContentPutInternal(RunData data, Context context, Properties reqP
 		String id = reqProps.getProperty(LTIService.LTI_ID);
 		String toolId = reqProps.getProperty(LTIService.LTI_TOOL_ID);
 
+		// Encrypt the lti_content secret(SAK-46566)
+		String newSecret = reqProps.getProperty(LTIService.LTI_SECRET);
+		if (LTIService.SECRET_HIDDEN.equals(newSecret)) {
+			reqProps.remove(LTIService.LTI_SECRET);
+			newSecret = null;
```
Comment (ern) [inline basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:1463]: setting this to null is seems redundant

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java [hunk 2]
```diff
@@ -1452,6 +1456,15 @@ private void doContentPutInternal(RunData data, Context context, Properties reqP
...
+		} else if (newSecret != null && newSecret.length() > 0) {
```
Comment (ern) [inline basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:1464]: StringUtils.isNotBlank(newSecret) is a better choice here

File: basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java (hunk 1 above)
Comment (kunaljaykam) [reply basiclti/basiclti-tool/src/java/org/sakaiproject/blti/tool/LTIAdminTool.java:1463]: I agree, Should we change this line of code too then?

`	// Insert or edit
	public void doToolPut(RunData data, Context context) {
		String peid = ((JetspeedRunData) data).getJs_peid();
		SessionState state = ((JetspeedRunData) data).getPortletSessionState(peid);

		if (!ltiService.isMaintain(getSiteId(state))) {
			addAlert(state, rb.getString("error.maintain.delete"));
			switchPanel(state, "Error");
			return;
		}
		Properties reqProps = data.getParameters().getProperties();

		String newSecret = reqProps.getProperty(LTIService.LTI_SECRET);
		if (LTIService.SECRET_HIDDEN.equals(newSecret)) {
			reqProps.remove(LTIService.LTI_SECRET);
			newSecret = null;
		}`

**Review by ern — APPROVED** [review]

**Pull request merged**